xbmc/music/MusicDatabase.cpp

```diff
@@ -10683,12 +10704,14 @@ bool CMusicDatabase::GetArtForItem(int songId, int albumId, int artistId, bool b
         MediaTypeAlbum, songId);
       strSQL = strSQL + " UNION " + strSQL2;
     }
-
     m_pDS2->query(strSQL);
     while (!m_pDS2->eof())
     {
       ArtForThumbLoader artitem;
       artitem.artType = m_pDS2->fv("type").get_asString();
+      if (discId > 0)
+        if (artitem.artType == StringUtils::Format("discart%i", discId))
+          artitem.artType = "discart";
```

Comment:
Need to handle any type of disc specific art not just discart. OK for now that is the only thing available, but avoidng hard coded art types in core. 
But anyway by doing all the art handling in `CMusicThumbLoader::FillLibraryArt` can avoid adding discid parameter